erpnext/buying/doctype/purchase_common/purchase_common.js

```diff
@@ -221,6 +221,42 @@ erpnext.buying.BuyingController = erpnext.TransactionController.extend({
 
 	tc_name: function() {
 		this.get_terms();
+	},
+	link_requests: function() { 
+		
+		cur_frm.add_custom_button(__("Link to Material Requests"),
+			function() {
+				my_items = [];
+				for (var i in cur_frm.doc.items) {
+					if(!cur_frm.doc.items[i].material_request){
+						my_items.push(cur_frm.doc.items[i].item_code);
+					}
+					
+				}
+				frappe.call({	
+					method: "erpnext.buying.doctype.purchase_common.purchase_common.get_linked_material_requests", 
+					args:{
+						items: my_items						
+					}, 
+					callback: function(r) { 
+						for (var i in cur_frm.doc.items) {
+							var qty = cur_frm.doc.items[i].qty;
+							(r.message || []).forEach(function(d) {
+								if (d[0].qty > 0 && qty > 0 && cur_frm.doc.items[i].item_code == d[0].item_code)
```

Comment:
I don't think that would be a problem based on the code here. The nested loop goes through each of the items in the item table and then loops through all the material requests. If the same item shows up multiple times, the quantities already assigned to the first instance of that item are accounted for and the MR used has it's quantity already reduced.